src/System.Memory/tests/ReadOnlySpan/LastIndexOf.char.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.SpanTests
```

Comment:
It might be worth adding tests that use a reference type T implementing IEquatable. I think all the current tests would work regardless of the code using IEquatable or simply comparing bits in the structs.

Reply:
Added tests for `{RO}Span<string>`

Reply:
Well, it's better than just testing the structs, but unfortunately String.Equals(string) does ordinal comparison, i.e. compares bits in the string :-). Maybe it's ok though. 

Also, I just realized we should probably add LastIndexOf(T, Comparer<T>) to let people chose whatever string comparison they want (not this PR of course).